### PR TITLE
chore: BaseInput props comments

### PR DESCRIPTION
**Pull request by connoratrug**
add comments describing the BaseInput props comments and there relation to emx2 fields

### CODE REVIEW

File: apps/molgenis-components/src/components/forms/BaseInput.vue [hunk 1]
```diff
@@ -2,24 +2,44 @@
 export default {
   name: 'BaseInput',
   props: {
+    /**
+     * Unique identifier https://developer.mozilla.org/en-US/docs/Web/HTML/Global_attributes/id
+     * Has no direct emx2 implementenation but can be constructend by concatinating the table name with the field name
+     */
```
Comment (mswertz) [inline apps/molgenis-components/src/components/forms/BaseInput.vue:8]: should we then do this automatically if 'null'? Or how do you recommend us to use this?

File: apps/molgenis-components/src/components/forms/BaseInput.vue [hunk 2]
```diff
@@ -2,24 +2,44 @@
...
+     * relates to emx 2 columnName
+     */
     label: {
       type: String,
       required: false
     },
+    /**
+     * Additional information describing the field 
+     * relates to emx 2 description
+     */
```
Comment (mswertz) [inline apps/molgenis-components/src/components/forms/BaseInput.vue:34]: why don't we then call it 'description'?

File: apps/molgenis-components/src/components/forms/BaseInput.vue [hunk 3]
```diff
@@ -2,24 +2,44 @@
...
     helpText: {
       type: String,
       required: false
     },
+    /**
+     * A string specifying the type of control to render. 
+     * (note: no direct relation to emx2 type, emx2 types are implemented as vue components)
+     */
     type: {
```
Comment (mswertz) [inline apps/molgenis-components/src/components/forms/BaseInput.vue:43]: so this is not related to columnType?

**Review by mswertz — COMMENTED** [review]
I have added a few questions.

File: apps/molgenis-components/src/components/forms/BaseInput.vue (hunk 1 above)
Comment (connoratrug) [reply apps/molgenis-components/src/components/forms/BaseInput.vue:8]: it can be done 'automatically' in a higher level comp that extends or is composed using the baseComp. The usage context can 'generate' the id avoid the need for the component to generate a random id

File: apps/molgenis-components/src/components/forms/BaseInput.vue [hunk 4]
```diff
@@ -2,24 +2,44 @@
...
+     * Has no direct emx2 implementenation but can be constructend by concatinating the table name with the field name
+     */
     id: {
       type: String,
       required: true
     },
+    /**
+     * Name of the element. For example used by the server to identify the fields in form submits.
+     * relates to emx 2 columnName
+     */
```
[the remark this answers is not in the document]
Comment (connoratrug) [reply apps/molgenis-components/src/components/forms/BaseInput.vue:16]: The name is more restrictive and used for communicating with the backend, the label can fallback to the name but should be focused on the human user. Not all the fields are generated form the schema, but all schema fields can be represented as a input ( except of heading :( )

File: apps/molgenis-components/src/components/forms/BaseInput.vue (hunk 2 above)
Comment (connoratrug) [reply apps/molgenis-components/src/components/forms/BaseInput.vue:34]: We use bootstap as  a htm css basis for our components , bootstrap calls this 'help' , imo its also more semantic as the label should describe the field and the help text can optionally add some help where needed ( like a comment). we could call it description. probably the route cause lies with emx mixing the 'desciption' and 'identified' into a single name  field and then needing a description to describe the field

File: apps/molgenis-components/src/components/forms/BaseInput.vue (hunk 3 above)
Comment (connoratrug) [reply apps/molgenis-components/src/components/forms/BaseInput.vue:43]: correct, the base component type allows the extending types to set the html input type attr , i would suggest having vue components that have a ( non 1-to-1) mapping with emx fields ( like implemented in styleguide dist). at the html level there is still the need to use the html type attr to communicate with the browsers. ( example , the InputString uses the BaseInput.type to set the html input tag to type="text")

**Comment by connoratrug** [comment]
answered the questions, note; This implementation of a base component does not really add any new fields , it moves the implementation of the id field outside of the baseComponent ( to allow for use in ssr context). Label , Name and Placeholder can still link to the same emx2 field but are explicitly defined and can differ ( when used directly in app instead of being generated from table to form)

File: apps/molgenis-components/src/components/forms/BaseInput.vue (hunk 1 above)
Comment (mswertz) [reply apps/molgenis-components/src/components/forms/BaseInput.vue:8]: What intermediate do you think of? Because I would suppose we don't want to repeat this for every *Input? I.e. StringInput, IntInput etc?

File: apps/molgenis-components/src/components/forms/BaseInput.vue (hunk 4 above)
[the remark this answers is not in the document]
Comment (mswertz) [reply apps/molgenis-components/src/components/forms/BaseInput.vue:16]: ok. EMX2 actually provides 'form elements' hence the 'heading'. But on topic: in most cases we now use v-model right? So I try to understand where we use the 'name'? Or is this a more theoretical consideration?

File: apps/molgenis-components/src/components/forms/BaseInput.vue (hunk 2 above)
Comment (mswertz) [reply apps/molgenis-components/src/components/forms/BaseInput.vue:34]: please call it 'description'. Only alternative is that we discuss with DM if we gonna rename 'description' to 'help' but I don't think that is an improvement

NB our focus is not 'bootstrap' but 'emx2' so we try to adapt bootstrap to emx2 instead of adapting emx2 to bootstrap. 
NB2 'description' is not the same a 'name' so it seems we have different understanding of emx2 metadata definition.  If you want to say something about emx 'mixing' then it is that emx2 tries to stop confusion between different 'name', 'identifier' and 'label' for same field, by saying you can have only one 'name' for a field. 'description' has the purpose to attach a text length explanation to that field. This could be in the form of a question, or in the form of an explanation.

File: apps/molgenis-components/src/components/forms/BaseInput.vue (hunk 3 above)
Comment (mswertz) [reply apps/molgenis-components/src/components/forms/BaseInput.vue:43]: ok, that makes sense.

**Comment by mswertz** [comment]
> answered the questions, note; This implementation of a base component does not really add any new fields , it moves the implementation of the id field outside of the baseComponent ( to allow for use in ssr context). Label , Name and Placeholder can still link to the same emx2 field but are explicitly defined and can differ ( when used directly in app instead of being generated from table to form)

Thanks. My tension is that we make code unnecessarily complex by requiring all these properties to be set. Base philosophy of emx2 styleguide is to provide building blocks fitting emx2. So I would really like to to (a) on level of StringInput expose only require things available in EMX2 metadata and (b) have (automatic) defaults for props that you might want to set in exceptional cases.

File: apps/molgenis-components/src/components/forms/BaseInput.vue (hunk 1 above)
Comment (connoratrug) [reply apps/molgenis-components/src/components/forms/BaseInput.vue:8]: what 'intermediate' or 'repeat' do you mean, could you be more specific or supply an example ?

File: apps/molgenis-components/src/components/forms/BaseInput.vue (hunk 4 above)
[the remark this answers is not in the document]
Comment (connoratrug) [reply apps/molgenis-components/src/components/forms/BaseInput.vue:16]: v-model is used for binding the input value to the js(-model), the name is used for html form element. when using js to post js model data to the backend named form elements are not needed, would still suggest to keep it to have valid html forms. the name is also use by browsers for validation and tabbing details https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input

**Comment by connoratrug** [comment]
> have (automatic) defaults for props that you might want to set in exceptional cases.

it has

> StringInput expose only require things available in EMX2 metadata

emx does not supply a page scoped id ( as it has no concept of a page or app) , or do suggest limiting all pages to a single table form ?

File: apps/molgenis-components/src/components/forms/BaseInput.vue (hunk 1 above)
Comment (mswertz) [reply apps/molgenis-components/src/components/forms/BaseInput.vue:8]: sorry, I meant to ask: what do you mean by 'higher level comp' ? is that something between 'BaseInput' and 'StringInput'?

File: apps/molgenis-components/src/components/forms/BaseInput.vue (hunk 4 above)
[the remark this answers is not in the document]
Comment (mswertz) [reply apps/molgenis-components/src/components/forms/BaseInput.vue:16]: yes, I get that (well, post that). But until now we never used 'name' so that is where my question came from :-). I don't mind this as long as we don't duplicate efforts. So defaulting this to the same as 'label' does the job

File: apps/molgenis-components/src/components/forms/BaseInput.vue (hunk 1 above)
Comment (connoratrug) [reply apps/molgenis-components/src/components/forms/BaseInput.vue:8]: EMXFormComponent or just FormComponent if you like
